File: String/leetcode/easy/LargestOddNumberInString.go
// https://leetcode.com/problems/largest-odd-number-in-string/

package main

import "fmt"

func largestOddNumber(num string) string {
    for i := len(num) - 1; i >= 0; i-- {
		if (num[i] - '0') % 2 != 0 {
			return num[:i+1]
		}
	}
    return ""
}

func largestOddNumberHelper() {
    testCases := []string{
        "52",
        "4206",
        "35427",
    }

    for i, testCase := range testCases {
        result := largestOddNumber(testCase)
        fmt.Printf("Test Case %d: %s\n", i+1, result)
    }
	fmt.Println()
}

File: String/leetcode/easy/LargestOddNumberInString.py
# https://leetcode.com/problems/largest-odd-number-in-string/

def largest_odd_number(num):
    for i in range(len(num) - 1, -1, -1):
        if int(num[i]) % 2 != 0:
            return num[:i + 1]
    return ""

# Example usage
if __name__ == "__main__":
    test_cases = [
        "52",
        "4206",
        "35427"
    ]
    
    for i, test_case in enumerate(test_cases, 1):
        result = largest_odd_number(test_case)
        print(f"Test Case {i}: {result}")
    